internal/pkg/cli/env_upgrade.go

```diff
@@ -212,6 +224,44 @@ func (o *envUpgradeOpts) listEnvsToUpgrade() ([]*config.Environment, error) {
 	return envs, nil
 }
 
+func (o *envUpgradeOpts) ensureManagerRoleIsAllowedToUpload(env *config.Environment, bucketName string) error {
+	cfn, err := o.newEnvDeployer(env)
+	if err != nil {
+		return err
+	}
+	body, err := cfn.EnvironmentTemplate(env.App, env.Name)
+	if err != nil {
+		return err
+	}
```

Comment:
If we are not using `cfn` and `body` in TODO, maybe we could move it into `isManagerRoleAllowedToUpload`?

Reply:
We will use it 👍